### PR TITLE
restructure controller context

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/controller/v24/controllercontext/status.go [hunk 1]
```diff
@@ -1,25 +1,22 @@
 package controllercontext
 
-type Status struct {
-	// Cluster carries information between cluster controller resource.
-	Cluster Cluster
-	// Drainer carries information between drainer controller resource.
-	Drainer Drainer
+type ContextStatus struct {
+	ControlPlane  ContextStatusControlPlane
+	TenantCluster ContextStatusTenantCluster
```
Comment (xh3b4sd) [inline service/controller/v24/controllercontext/status.go]: I would like to have this separation. One reason is the AWS account ID. I plan to improve that handling but the controller context needs to get aligned first.

File: service/controller/v24/controllercontext/status.go [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
-type Cluster struct {
-	AWSAccount             ClusterAWSAccount
-	ASG                    ClusterASG
+type ContextStatusControlPlane struct {
+	AWSAccountID string
+}
+
+type ContextStatusTenantCluster struct {
+	AWSAccountID           string
+	TCCP                   ContextStatusTenantClusterTCCP
```
Comment (xh3b4sd) [inline service/controller/v24/controllercontext/status.go]: For node pools we need to start differentiating between different stacks. Here is the stack reflected with the resource name. Later we add TCDP which will be the Tenant Cluster Data Plane in which the node pool stacks are managed.

File: service/controller/v24/controllercontext/status.go [hunk 3]
```diff
@@ -1,25 +1,22 @@
 package controllercontext
 
-type Status struct {
-	// Cluster carries information between cluster controller resource.
```
Comment (xh3b4sd) [inline service/controller/v24/controllercontext/status.go]: The controller separation on this level does not work. We have the workerasgname resource run in all controllers.

File: service/controller/v24/controllercontext/cluster_asg.go [hunk 4]
```diff
@@ -1,11 +1,12 @@
 package controllercontext
 
-type ClusterASG struct {
+type ContextStatusTenantClusterTCCPASG struct {
```
Comment (xh3b4sd) [inline service/controller/v24/controllercontext/cluster_asg.go]: The names do not get prettier but this is just how it is I guess.

Comment (rossf7) [reply service/controller/v24/controllercontext/cluster_asg.go]: Oof that is touch to read here. But when its used `cc.Status.TenantCluster.TCCP.ASG.Name` I'm fine with it.

File: service/controller/v24/controllercontext/status.go (hunk 1 above)
Comment (rossf7) [reply service/controller/v24/controllercontext/status.go]: Having the separation makes sense to me.

**Pull request merged**